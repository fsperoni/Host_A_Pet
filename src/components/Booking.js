import React, { useContext, useEffect, useState } from 'react';
import UserContext from '../hooks/useUserContext';
import HostAPetApi from './Api';
import AvailabilitySearchForm from './AvailabilitySearchForm';
import BookingCard from './BookingCard';
import '../styles/Booking.scss'

const Booking = () => {
  const { currentUser } = useContext(UserContext);
  const [avails, setAvails] = useState([]);
  const [roles, setRoles] = useState([]);

  useEffect(function getRoles() {
    const getAllRoles = async () => {
      const allRoles = await HostAPetApi.getAllRoles();
      setRoles(allRoles);
    }
    getAllRoles();
  }, [currentUser.username]);

  const bookingCards = avails.map(a => (
    <BookingCard key={a.id} avail={a} />
  ));

  return (
    <>
    <div className="Booking col-md-6 offset-md-3 col-lg-4 offset-lg-4">
      <h2 className="text-center">New Booking</h2>
      <AvailabilitySearchForm avails={avails} setAvails={setAvails} roles={roles} />
    </div>
    <div className="container col-md-6 mb-3">
      {avails.length ? 
        <>
        <div className="Booking-filter">
          <h4>Max distance (km):</h4>
          <select className="ms-2" name="distance">
            <option value="">Select one</option>
            <option value="5">5</option>
            <option value="10">10</option>
            <option value="20">15</option>
            <option value="50">50</option>
            <option value="100">100</option>
          </select>
          <button className="btn btn-primary btn-block ms-3">
            Go
          </button>
        </div>
        <div className="d-grid mt-4">{bookingCards}</div>
        </> :
        <h2 className="mt-4 text-center">Nothing to show yet!</h2>
      }
    </div>
    </>
  )
}

export default Booking;