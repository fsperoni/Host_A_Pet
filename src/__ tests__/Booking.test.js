import { render } from "@testing-library/react";
import Booking from "../components/Booking";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser,
} = require("./_testCommon");

// smoke test
it('renders without crashing', function () {
  render(
    <UserContext.Provider
      value={{ currentUser: currentAdminUser(), function() { } }}>
      <Booking />
    </UserContext.Provider>
  );
});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <UserContext.Provider
      value={{ currentUser: currentAdminUser(), function() { } }}>
      <Booking />
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'New Booking' in the component.", function () {
  const { getByText } = render(
    <UserContext.Provider
      value={{ currentUser: currentAdminUser(), function() { } }}>
      <Booking />
    </UserContext.Provider>
  );
  expect(getByText("New Booking")).toBeInTheDocument();
});