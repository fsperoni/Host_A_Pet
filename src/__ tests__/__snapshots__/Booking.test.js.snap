// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Booking col-md-6 offset-md-3 col-lg-4 offset-lg-4"
  >
    <h2
      class="text-center"
    >
      New Booking
    </h2>
    <div
      class="AvailabilityAddForm mt-3"
    >
      <div
        class="container"
      >
        <h3
          class="mb-3"
        >
          Start your search!
        </h3>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <form>
              <div
                class="form-group"
              >
                <label>
                  Looking for a:
                </label>
                <select
                  class="form-select"
                  name="roleId"
                >
                  <option
                    value=""
                  >
                    Select one
                  </option>
                </select>
              </div>
              <div
                class="form-group"
              >
                <label>
                  From:
                </label>
                <input
                  class="form-control"
                  min="2022-04-02"
                  name="startDate"
                  type="date"
                  value="2022-04-02"
                />
              </div>
              <div
                class="form-group"
              >
                <label>
                  Until:
                </label>
                <input
                  class="form-control"
                  min="2022-04-02"
                  name="endDate"
                  type="date"
                  value="2022-04-02"
                />
              </div>
              <div
                class="container mt-4"
              >
                <button
                  class="btn btn-primary btn-block"
                >
                  Search
                </button>
                <button
                  class="btn btn-warning btn-block ms-4"
                >
                  Cancel
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container col-md-6 mb-3"
  >
    <h2
      class="mt-4 text-center"
    >
      Nothing to show yet!
    </h2>
  </div>
</DocumentFragment>
`;
